### PR TITLE
continuous model first commit

### PR DESCRIPTION
**Pull request by JEN6YT**

### CODE REVIEW

File: Model/ctpm.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+    Returns:
+    - obj: the computed objective.
+    """
+    optimizer = Adam(model.parameters(), lr=lr)
+    optimizer.zero_grad()
+
+    for epoch in range(epochs):
+        obj, a, b, _, _ = model(D_tre, D_unt, o_tre, o_unt, c_tre, c_unt, i_tre, i_unt)
+        (-obj).backward()  # Negative objective for maximization
```
Comment (wzoustanford) [inline Model/ctpm.py:169]: In the model objective, it's c / o. This line says it's maximizing c / o :)

File: Model/ctpm.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        h_tre_rnkscore, h_unt_rnkscore = s_tre.clone(), s_unt.clone()
+
+        # Top-k sorting with differentiable approximation
+        top_k_tre = int(torch.ceil(torch.tensor(size_tre * self.p_quantile)).item())
+        top_k_unt = int(torch.ceil(torch.tensor(size_unt * self.p_quantile)).item())
+
+        h_tre_sorted, _ = torch.sort(s_tre, dim=0, descending=True)
+        h_unt_sorted, _ = torch.sort(s_unt, dim=0, descending=True)
+
+        intercept_tre = h_tre_sorted[top_k_tre - 1].detach()
```
Comment (wzoustanford) [inline Model/ctpm.py]: Let's try the continuous treatment ranking model without the quantile/constrained ranking, first. 

The constrained ranking may not be important to combine with continuous treatment ranking model.

File: Model/ctpm.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        intercept_unt = h_unt_sorted[top_k_unt - 1].detach()
+
+        h_tre = torch.sigmoid(self.temp * (s_tre - intercept_tre))
+        h_unt = torch.sigmoid(self.temp * (s_unt - intercept_unt))
+
+        h_tre = F.dropout(h_tre, self.dropout_rate)
```
Comment (wzoustanford) [inline Model/ctpm.py]: I also couldn't recall the details of applying dropout. Feel free to try it out but start from dropout_rate = 0.0
